subcommands/version/updates.go

```diff
@@ -158,11 +158,7 @@ func (u FioctlUpdate) Do() error {
 	}
 
 	fmt.Println("Saving new version to", exe)
-	tmp := exe + ".tmp"
-	if err = os.WriteFile(tmp, pb.buff.Bytes(), st.Mode()); err != nil {
-		return err
-	}
-	return os.Rename(tmp, exe)
+	return updateSelf(exe, pb.buff.Bytes(), st.Mode())
```

Comment:
I _believe_ the tempfile writing needs to be a common part.

Now, after some thought I think that _maybe_ writing to `/tmp` is not that good after all, as it may be on the other partition, making the rename non-atomic.
Then, adding _nonse_ does not make sense, as it may lead to garbage in `/usr/bin`.
So your original idea with just adding the `.tmp` prefix winds up to be the best choice for us.

I leave it up to you to decide. I guess you've gave it a good thought.

Reply:
its not great but I landed on this and the other tools I see in this realm do it also.

Reply:
I wonder if this "self-updating" will work if `fioctl` is a link?
E.g. `/usr/bin/fioctl --> /myplace/fioctl` will `os.Executable` return `/myplace/fioctl`?

Reply:
> I wonder if this "self-updating" will work if fioctl is a link?

It does. This is my home set up. I had to set things up to test the "normal" path where its not a link